### PR TITLE
Fix unchecked malloc

### LINKED ISSUE #5428: VeraCode Unchecked Error Condition FIRCLSReportAdapter.m
**Environment**

Xcode version: 11.3.1
Firebase SDK version: 6.21.0
Firebase Component: Crashlytics
Installation method: CocoaPods 

**Description:** 

The result of this call to malloc() is not checked for success before being used. This can result in application instability or crashing if memory is not available.

**Relevant Code:**

../Record/FIRCLSReportAdapter.m 159

 .../Record/FIRCLSReportAdapter.m 234

### PR DESCRIPTION
**Pull request by morganchen12**
Fixes #5428.

### CODE REVIEW

File: Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m [hunk 1]
```diff
@@ -155,19 +155,21 @@ - (google_crashlytics_FilesPayload)protoFilesPayload {
...
-    google_crashlytics_FilesPayload_File file = google_crashlytics_FilesPayload_File_init_default;
-    file.filename = FIRCLSEncodeString(clsRecords[i].lastPathComponent);
-
-    NSError *error;
-    file.contents = FIRCLSEncodeData([NSData dataWithContentsOfFile:clsRecords[i]
-                                                            options:0
-                                                              error:&error]);
-    if (error) {
-      FIRCLSErrorLog(@"Failed to read from %@ with error: %@", clsRecords[i], error);
+  if (files != NULL) {
```
Comment (samedson) [inline Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m:158]: More of a style thing, but I like to keep the happy path code in the least-indent, and use if statements to check for issues. Can we change this to:

if (files == NULL) {
  return apple_payload;
}

File: Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m [hunk 2]
```diff
@@ -230,8 +232,10 @@ - (google_crashlytics_Platforms)protoPlatformFromString:(NSString *)str {
  */
 pb_bytes_array_t *FIRCLSEncodeData(NSData *data) {
   pb_bytes_array_t *pbBytes = malloc(PB_BYTES_ARRAY_T_ALLOCSIZE(data.length));
-  memcpy(pbBytes->bytes, [data bytes], data.length);
-  pbBytes->size = (pb_size_t)data.length;
+  if (pbBytes != NULL) {
+    memcpy(pbBytes->bytes, [data bytes], data.length);
+    pbBytes->size = (pb_size_t)data.length;
+  }
```
Comment (samedson) [inline Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m:238]: Same here - can we just return NULL when pbBytes == null?

Comment (morganchen12) [reply Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m:239]: Done.

File: Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m (hunk 1 above)
Comment (morganchen12) [reply Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m]: Done.

File: Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m (hunk 2 above)
Comment (morganchen12) [reply Crashlytics/Crashlytics/Models/Record/FIRCLSReportAdapter.m:239]: Done.

**Pull request merged**